src/bindings/html/service.js

```diff
@@ -1,21 +1,68 @@
 'use strict';
 
-import { translateDocument, dispatchEvent } from './dom';
+import Env from '../../lib/env';
+import { View, init as initView } from './view';
+import { getMeta } from './head';
+import { getAdditionalLanguages, changeLanguage } from './langs';
 
 export const L10n = {
   views: [],
   env: null,
   languages: null,
-  change: null
+  requestLanguages: null
 };
 
 export function initViews(langs) {
   return Promise.all(
-    this.views.map(view => initView(view, langs)));
+    this.views.map(view => initView.call(view, langs)));
 }
 
-function initView(view, langs) {
-  dispatchEvent(view.doc, 'supportedlanguageschange', langs);
-  return view.ctx.fetch(langs, 1).then(
-    translateDocument.bind(view, view.doc, langs));
+export function onlanguagechage(
+  appVersion, defaultLang, availableLangs, requestedLangs) {
+
+  return this.languages = Promise.all([
+    getAdditionalLanguages(), this.languages]).then(
+      ([additionalLangs, prevLangs]) => changeLanguage(
+        initViews.bind(this), appVersion, defaultLang, availableLangs,
+        additionalLangs, prevLangs, requestedLangs || navigator.languages));
+}
+
+export function onadditionallanguageschange(
+  appVersion, defaultLang, availableLangs, additionalLangs,
+  requestedLangs) {
+
+  return this.languages = this.languages.then(
+    prevLangs => changeLanguage(
+      initViews.bind(this), appVersion, defaultLang, availableLangs,
+      additionalLangs, prevLangs, requestedLangs || navigator.languages));
+}
+
+export function init(fetch, additionalLangsAtLaunch) {
```

Comment:
I want this to be a privileged method: our code can call it, but the consumer of the API cannot, because it makes little sense to init the `L10n` singleton more than once.  This is why I define it here as an exported function and bind it to `L10n` in the runtime.